### PR TITLE
Use 'override' whenever possible instead of 'virtual'

### PR DESCRIPTION
**Pull request by vittorioromeo**
## Description

Pure refactoring change. Mostly automated with `clang-tidy`, but also some manual work to fix-up some comments and some places where `clang-tidy` failed.

Context: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#c128-virtual-functions-should-specify-exactly-one-of-virtual-override-or-final

## Tasks

* [ ] Tested on Linux
* [x] Tested on Windows
* [ ] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

## How to test this PR?

CI. I've also ran some of the examples and everything was fine.

### CODE REVIEW

**Comment by vittorioromeo** [comment]
@Bromeon: wanna take a look at this if you have some time? Should be fairly straightforward

File: include/SFML/Audio/SoundFileReader.hpp [hunk 1]
```diff
@@ -4,11 +4,12 @@
...
 // This software is provided 'as-is', without any express or implied warranty.
-// In no event will the authors be held liable for any damages arising from the use of this software.
+// In no event will the authors be held liable for any damages arising from the
+// use of this software.
 //
 // Permission is granted to anyone to use this software for any purpose,
-// including commercial applications, and to alter it and redistribute it freely,
-// subject to the following restrictions:
+// including commercial applications, and to alter it and redistribute it
+// freely, subject to the following restrictions:
```
Comment (Bromeon) [inline include/SFML/Audio/SoundFileReader.hpp:12]: Why these line changes? Some accidental auto-formatting?

File: include/SFML/Audio/SoundFileReader.hpp [hunk 2]
```diff
@@ -132,26 +136,29 @@ class SFML_AUDIO_API SoundFileReader
 ///
 ///     static bool check(sf::InputStream& stream)
 ///     {
-///         // typically, read the first few header bytes and check fields that identify the format
+///         // typically, read the first few header bytes and check fields that
+///         identify the format
```
Comment (Bromeon) [inline include/SFML/Audio/SoundFileReader.hpp:140]: line change

File: include/SFML/Audio/SoundFileReader.hpp [hunk 3]
```diff
@@ -132,26 +136,29 @@ class SFML_AUDIO_API SoundFileReader
...
+///         sound
 ///     }
 ///
-///     virtual sf::Uint64 read(sf::Int16* samples, sf::Uint64 maxCount)
+///     sf::Uint64 read(sf::Int16* samples, sf::Uint64 maxCount) override
 ///     {
 ///         // read up to 'maxCount' samples into the 'samples' array,
-///         // convert them (for example from normalized float) if they are not stored
+///         // convert them (for example from normalized float) if they are not
+///         stored
```
Comment (Bromeon) [inline include/SFML/Audio/SoundFileReader.hpp:161]: line change

**Review by Bromeon — COMMENTED** [review]
Looks good! Just some unrelated line changes where I wasn't sure if they are intended.

Maybe in the future, we could enforce `override` in a CI lint?

File: include/SFML/Audio/SoundFileReader.hpp (hunk 3 above)
Comment (vittorioromeo) [reply include/SFML/Audio/SoundFileReader.hpp:161]: Will fix

File: include/SFML/Audio/SoundFileReader.hpp (hunk 2 above)
Comment (vittorioromeo) [reply include/SFML/Audio/SoundFileReader.hpp:140]: Will fix

File: include/SFML/Audio/SoundFileReader.hpp (hunk 1 above)
Comment (vittorioromeo) [reply include/SFML/Audio/SoundFileReader.hpp:12]: Yes, sorry! Will fix.

**Comment by vittorioromeo** [comment]
> Looks good! Just some unrelated line changes where I wasn't sure if they are intended.

Thanks. They should be fixed now.

> Maybe in the future, we could enforce `override` in a CI lint?

Absolutely, I will create an issue for it.

**Comment by vittorioromeo** [comment]
> As discussed [here](https://github.com/SFML/SFML-Buildbot/issues/28), `-Wsuggest-override` can/should be added to the set of warning flags in this PR.

I have performed the change and rebased, let's see if CI likes it.

**Review by Bromeon — APPROVED** [review]

**Pull request merged**